ca/keyreadwriter.go

```diff
@@ -240,13 +256,14 @@ func (k *KeyReadWriter) writeKey(keyBlock *pem.Block, writeKEK []byte, hm PEMKey
 
 	var onSuccess func()
 	if hm != nil {
-		headers, successFunc, err := hm.GetNewHeaders(writeKEK)
+		headers, successFunc, err := hm.GetNewHeaders(kekData)
 		if err != nil {
 			return err
 		}
 		mergePEMHeaders(keyBlock.Headers, headers)
 		onSuccess = successFunc
 	}
+	keyBlock.Headers[versionHeader] = strconv.FormatUint(kekData.Version, 10)
```

Comment:
If we store the version as `api.Version`, we could serialize the protobuf object into the header. Not sure if this is worth it though.